src/components/Navigation/Navigation.ts

```diff
@@ -37,6 +38,28 @@ const Navigation = (props: any, { slots, attrs }: any) => {
     return directionIcons[normalizeDir.value]
   }
 
+  const { width, height, borderRadius, color, colorHover, background } = props
+  const style = generateStyleVars('nav', {
```

Comment:
I'm not a fan of inlining styles as prop, I think the api with css vars is already sufficient and css vars should stay in css realm, not template or js component, if users really want to add them inline they can already do

`<Pagination :style="{'--vc-pgn-width': '20px', '--vc-pgn-border-radius': '5px'}" />`

So this is pretty redundant, I think we should just update the doc for this

Reply:
Thank you for sharing your thoughts! While I understand the preference for keeping CSS variables within CSS for clarity and separation of concerns, I'd like to propose an alternative approach that offers greater flexibility while maintaining an intuitive API and addressing users' common pain points.

I propose a more intuitive higher-level API inspired by JSS:

```vue
<CarouselPagination :custom-style="{width: 24, height: 10, borderRadius: 5, top: '10px'}" />
```

or, alternatively:

```vue
<CarouselPagination v-bind="{width: 24, height: 10, borderRadius: 5, top: '10px'}" />
```

This would provide

1. **Flexibility**: Developers who prefer complete control can still use the existing :style or global CSS variables approach.
2. **Addressing User Feedback**: This API addresses common frustrations raised in the community, as seen in #334 and #334.

Reply:
I would suggest using a prop like css-vars and not adding any client side validation or prop selection to avoid making this more bulky than it needs to be if you really want to add it. Only provide typescript types and then just output anything passed to the prop as css variables with the prefix. Units should also be explicit like in css. 

Assuming a number means pixel is never a good idea because of the amount of different units and because some variables will not need units like rgb colors or some variables might reference a time like 0.5s